### PR TITLE
GEODE-1296: change conditional in getRawBytes to assert

### PR DESCRIPTION
**Pull request by pdxrunner**
Made suggested change to getRawBytes. Updated corresponding unit test.
See review request https://reviews.apache.org/r/47479/

### CODE REVIEW

**Comment by dschneider-pivotal** [comment]
I think this should be changed to use the java "assert" keyword. The assertion should only be checked when assertions on the jvm are enabled.

File: geode-core/src/main/java/com/gemstone/gemfire/internal/offheap/OffHeapStoredObject.java [hunk 1]
```diff
@@ -417,11 +417,13 @@ public void setSerializedValue(byte[] value) {
       MemoryAllocatorImpl.getAllocator().getStats().incReads();
       return result;
     }
+    /**
+     * This method should only be called on uncompressed objects
+     * @return byte array of the StoredObject value. 
+     */
     protected byte[] getRawBytes() {
+      assert !isCompressed();
       byte[] result = getCompressedBytes();
```
Comment (dschneider-pivotal) [inline geode-core/src/main/java/com/gemstone/gemfire/internal/offheap/OffHeapStoredObject.java]: Combine these two lines into "return getCompressedBytes();"

File: geode-core/src/test/java/com/gemstone/gemfire/internal/offheap/OffHeapStoredObjectJUnitTest.java [hunk 2]
```diff
@@ -42,6 +42,10 @@
 @Category(UnitTest.class)
 public class OffHeapStoredObjectJUnitTest extends AbstractStoredObjectTestBase {
 
+  static {
```
Comment (dschneider-pivotal) [inline geode-core/src/test/java/com/gemstone/gemfire/internal/offheap/OffHeapStoredObjectJUnitTest.java]: Remove this static block. I talked with Kirk and we think that java assertions are enabled consistently from both gradle and the IDE

File: geode-core/src/test/java/com/gemstone/gemfire/internal/offheap/OffHeapStoredObjectJUnitTest.java [hunk 3]
```diff
@@ -816,6 +820,7 @@ public void getRawBytesShouldThrowExceptionIfValueIsCompressed() {
     chunk.getRawBytes();
 
     chunk.release();
+    fail("Expected getRawBytes() for a compressed value to throw java.lang.AssertionError");
```
Comment (dschneider-pivotal) [inline geode-core/src/test/java/com/gemstone/gemfire/internal/offheap/OffHeapStoredObjectJUnitTest.java]: Why is this "fail" call added. Doesn't the "expected" on the Test annotation take care of this?

**Comment by pdxrunner** [comment]
Precheckin passed with my most recent update. I think this one is ready to go now.

**Comment by dschneider-pivotal** [comment]
The travis failure is a know intermittent issue in a new unit test.
This change looks good. I will pull it in.

**Comment by pdxrunner** [comment]
This was checked in by dschneider. Closing the PR

**Pull request closed without merging**